### DISCUSSION TITLE
Is there any alternative to @jmandzik/apollo-server-plugin-server-timing that works with Apollo 3+?

### DISCUSSION THREAD

**Discussion by FezVrasta** [apollographql/apollo-server#7366] [General]
I use `@jmandzik/apollo-server-plugin-server-timing` on my Apollo 2 server, but trying to upgrade to the latest Apollo version I noticed the plugin is not compatible, is there any alternative available?

**Comment by trevor-scheer** [comment]
Hey @FezVrasta, I'm not aware of an alternative for v4. That said, the [usage reporting plugin](https://github.com/apollographql/apollo-server/blob/d057e2ffccac2afc9c3e102db64d74d895157c3d/packages/server/src/plugin/usageReporting/plugin.ts) collects timing data by implementing the `willResolveField` hook. You might be able to leverage the traces data it generates or use it as an example for your own theoretical plugin that collects this data.

**Reply by FezVrasta** [reply]
Thanks! Having this capability provided by Apollo itself would be great, it's a real good way to profile performance.

**Reply by glasser** [reply]
Alternatively you could have folks install the [inline trace plugin](https://www.apollographql.com/docs/apollo-server/api/plugin/inline-trace/) and parse the protobuf it adds to the response extensions.

**Reply by glasser** [reply]
(I agree that the built-in dev tools are a great tracing tool, but you can also use Explorer in Apollo Studio which lets you visualize inline traces.)
